importApp/src/test/java/com/commercetools/dataimport/joyrideavailability/AvailabilityPricesImportJobIntegrationTest.java

```diff
@@ -0,0 +1,141 @@
+            final Product productToDelete = sphereClient.executeBlocking(ProductUpdateCommand.of(updatedProduct, Unpublish.of()));
+            sphereClient.executeBlocking(ProductDeleteCommand.of(productToDelete));
+        });
+    }
+
+    @Test
+    public void jobAvailabilityPricesImport() throws Exception {
+        final int amountOfProducts = 10;
+        withJoyrideChannels(sphereClient, joyrideChannels -> {
+            withListOfProductProjections(sphereClient, amountOfProducts, productsWithoutJoyride -> {
+                assertThat(productsWithoutJoyride.size()).isEqualTo(amountOfProducts);
+                executeAvailabilityPricesImportJob();
+                final List<ProductProjection> updatedProducts = sphereClient.executeBlocking(ProductProjectionQuery.ofCurrent().withExpansionPaths(m -> m.allVariants().prices().channel())).getResults();
+                updatedProducts.forEach(m -> {
+                    assertThat(productContainJoyridePrice(m)).isTrue();
+                    final ProductProjection productProjectionWithoutJoyride = productsWithoutJoyride.stream().filter(p -> p.getId().equals(m.getId())).limit(1L).collect(toList()).get(0);
+                    final Price originalMasterVariantPrice = productProjectionWithoutJoyride.getMasterVariant().getPrices().get(0);
+                    final ProductVariant masterVariant = m.getMasterVariant();
+                    masterVariant.getPrices().forEach(variantPrice -> {
+                        if (variantPrice.getChannel() != null) {
+                            final PriceDraftDsl randomPriceDraft = randomPriceDraft(masterVariant, variantPrice.getChannel().getObj(), originalMasterVariantPrice);
+                            assertThat(variantPrice.getValue()).isEqualTo(randomPriceDraft.getValue());
+                        }
+                    });
+                });
+                return updatedProducts;
+            });
+        });
+    }
+
+    @Test
+    public void readerRestartAfterPartialExecution() throws Exception {
+        final int amountOfProducts = 10;
+        final int stopIndex = 4;
+        withPriceWithJoyrideChannel(sphereClient, priceDraft -> {
+            withListOfProductProjections(sphereClient, amountOfProducts, productsWithoutJoyride -> {
+                final ItemReader<ProductProjection> initialReader = createReader(sphereClient);
+                addJoyridePriceToProductsUntilStopIndex(stopIndex, priceDraft, initialReader);
+                final ItemReader<ProductProjection> restartReader = createReader(sphereClient);
+                addJoyrideToMissingProducts(stopIndex, priceDraft, productsWithoutJoyride, restartReader);
+                final List<ProductProjection> products = sphereClient.executeBlocking(ProductProjectionQuery.ofCurrent().withExpansionPaths(m -> m.allVariants().prices().channel())).getResults();
+                products.forEach(m -> assertThat(productContainJoyridePrice(m)).isTrue());
```

Comment:
This assertion relies very much on the production code.
So if in theory I would implement `productContainJoyridePrice` to just return `true` then the test would still be green?

It would be safer if you query your 10 products with a price selection and check that the right price is present in the `price` field.